File: Data Structures & Algorithms/implement-stack-using-queues/submission-0.swift
class MyStack {

    var queue: [Int] = []

    init() {

    }

    func push(_ x: Int) {
        queue.append(x)
        // rotate
        for i in 0 ..< queue.count - 1 {
            queue.append(queue.removeFirst())
        }
    }

    func pop() -> Int {
        return queue.removeFirst()
    }

    func top() -> Int {
        return queue.first!
    }

    func empty() -> Bool {
        return queue.count == 0
    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * let obj = MyStack()
 * obj.push(x)
 * let param_2 = obj.pop()
 * let param_3 = obj.top()
 * let param_4 = obj.empty()
 */

File: Data Structures & Algorithms/logger-rate-limiter/submission-0.py
class Logger:

    def __init__(self):
        self.history = {} # {message: timestamp}
        

    def shouldPrintMessage(self, timestamp: int, message: str) -> bool:
        if message not in self.history or timestamp - self.history[message] >= 10:
            self.history[message] = timestamp
            return True
        return False
        
        


# Your Logger object will be instantiated and called as such:
# obj = Logger()
# param_1 = obj.shouldPrintMessage(timestamp,message)
